Ira/my-app-video/src/components/layout/Layout.tsx

```diff
@@ -0,0 +1,237 @@
+import {IFilm} from "../../interfaces/IFilm";
+import {ILayoutState} from "../../interfaces/ILayoutState";
+import './styles.scss';
+
+const emptyObjectFilm: IFilm = {
+  id: 0,
+  title: '',
+  tagline: '',
+  vote_average: 0,
+  vote_count: 0,
+  release_date: '',
+  poster_path: '',
+  overview: '',
+  budget: 0,
+  revenue: 0,
+  runtime: 0,
+  genres: [''],
+}
+
+let films: IFilm[];
+films = [
+  {
+    id: 1,
+    title: "kill bill",
+    tagline: 'Grammy',
+    vote_average: 0,
+    vote_count: 0,
+    release_date: '2006',
+    poster_path: "https://i.ytimg.com/vi/huM6iOVmeZI/maxresdefault.jpg",
+    overview: 'This is (IMO) a much better solution than the accepted answer, because it lets you define a type and then, say, pass a parameter of that type (the callback) which you can then use any way you want, including calling it. The accepted answer uses a member variable and you have to set the member variable to your function, then call a method - ugly and prone to errors, because setting the variable first is part of the contract of calling the method.  ',
+    budget: 0,
+    revenue: 0,
+    runtime: 0,
+    genres: ['action'],
+  },
+  {
+    id: 2,
+    title: "island",
+    tagline: 'Oscar',
+    vote_average: 0,
+    vote_count: 0,
+    release_date: '2006',
+    poster_path: "https://upload.wikimedia.org/wikipedia/ru/7/72/Fantasy_Island_%28film%2C_2020%29.jpg",
+    overview: '',
+    budget: 0,
+    revenue: 0,
+    runtime: 0,
+    genres: ['action', 'detective'],
+  },
+  {
+    id: 3,
+    title: "kill bill",
+    tagline: 'Oscar',
+    vote_average: 0,
+    vote_count: 0,
+    release_date: '2006',
+    poster_path: "https://i.ytimg.com/vi/huM6iOVmeZI/maxresdefault.jpg",
+    overview: '',
+    budget: 0,
+    revenue: 0,
+    runtime: 0,
+    genres: ['action'],
+  },
+  {
+    id: 4,
+    title: "kill bill",
+    tagline: '',
+    vote_average: 0,
+    vote_count: 0,
+    release_date: '2006',
+    poster_path: "https://i.ytimg.com/vi/huM6iOVmeZI/maxresdefault.jpg",
+    overview: '',
+    budget: 0,
+    revenue: 0,
+    runtime: 0,
+    genres: ['action'],
+  },
+  {
+    id: 5,
+    title: "kill bill",
+    tagline: '',
+    vote_average: 0,
+    vote_count: 0,
+    release_date: '2006',
+    poster_path: "https://i.ytimg.com/vi/huM6iOVmeZI/maxresdefault.jpg",
+    overview: '',
+    budget: 0,
+    revenue: 0,
+    runtime: 0,
+    genres: ['action'],
+  },
+  {
+    id: 6,
+    title: "kill bill",
+    tagline: '',
+    vote_average: 0,
+    vote_count: 0,
+    release_date: '2006',
+    poster_path: "https://i.ytimg.com/vi/huM6iOVmeZI/maxresdefault.jpg",
+    overview: '',
+    budget: 0,
+    revenue: 0,
+    runtime: 0,
+    genres: ['action'],
+  },
+];
+
+let film: IFilm[];
+film = [];
+
+export class Layout extends React.Component<{}, ILayoutState> {
+  state: ILayoutState = {
+    films: [...films],
+    isFullFilmInfo: false,
+    currentFilmId: '',
+    currentFilm: emptyObjectFilm,
```

Comment:
Are you sure you need all these data in state? Maybe current film id will be enough ? If there is an id, it means that you have to show film info, needed film you can pass from parent component 
films[id].